boxsdk/auth/oauth2.py

```diff
@@ -42,6 +43,10 @@ def __init__(
             Optional callback for getting access to tokens for storing them.
         :type store_tokens:
             `callable`
+        :param retrieve_tokens:
+            Optional callback for retrieving tokens prior to refresh.
```

Comment:
Let's expand on this slightly.

We should specify that the callback takes no arguments, and that it should either return None (if no credentials are currently stored), or a (access token, refresh token) tuple.

And maybe we should add a brief description of what this feature would be used for. Something like "When using multiple processes or multiple separate OAuth2 objects, provide a retrieve_tokens function so that the same credentials can be reused across all your OAuth2 objects. The first object that needs to authenticate will do so, and store the tokens with the store_tokens callback (or the authentication and storage might happen entirely outside of the SDK). All subsequent objects, rather than re-authenticating, will use the tokens provided by retrieve_tokens. And the same behavior will occur when they all need to do token refresh." Feel free to reword or make that more concise.